pyroute2/fixtures/iproute.py

```diff
@@ -0,0 +1,523 @@
+        return self._address
+
+    @property
+    def netns(self) -> str:
+        '''Test interface netns.
+
+        A string name of the network namespace.'''
+        return self._netns
+
+
+class TestContext:
+    '''The test context.
+
+    Provided by `async_context` and `sync_context` fixtures.
+
+    Provides convenient shortcuts to RTNL API, the network namespace
+    name and the test interface spec.
+    '''
+
+    def __init__(
+        self, ipr: Union[IPRoute, AsyncIPRoute], test_link: TestInterface
+    ):
+        self._ipr = ipr
+        self._test_link = test_link
+
+    @property
+    def ipr(self) -> Union[IPRoute, AsyncIPRoute]:
+        '''RTNL API.
+
+        Return RTNL API instance, either IPRoute, or AsyncIPRoute.'''
+        return self._ipr
+
+    @property
+    def test_link(self) -> TestInterface:
+        '''Test interface spec.
+
+        Return `TestInterface` object for the test interface.'''
+        return self._test_link
+
+    @property
+    def netns(self) -> str:
+        '''Network namespace.
+
+        A string name of the network namespace.'''
+        return self.ipr.status['netns']
+
+
+@pytest.fixture(name='nsname')
+def _nsname() -> Generator[str]:
+    '''Network namespace.
+
+    * **Name**: nsname
+    * **Scope**: function
+
+    Create a unique network namespace and yield its name. Remove
+    the netns on cleanup.
+
+    It's safe to create and modify interfaces, addresses, routes etc.
+    in the test network namespace, as it is disconnected from the main
+    system, and the test cleanup will remove the namespace with all
+    its content.
+
+    Example usage:
+
+    .. testcode::
+
+        def test_list_interfaces(nsname):
+            subprocess.Popen(
+                ['ip', 'netns', 'exec', nsname, 'ip', 'link'],
+                stdout=subprocess.PIPE,
+            )
+            # ...
+
+    .. testcode::
+        :hide:
+
+        import subprocess
+        from unittest.mock import patch
+        from pyroute2 import netns
+
+        calls = 0
+        nsptr = None
+
+        with patch("subprocess.Popen"):
+            for nsname in test_fixture(
+                fixture=iproute._nsname,
+                scope='function',
+                name='nsname'
+            ):
+                nsptr = nsname
+                calls += 1
+                test_list_interfaces(nsname)
+
+        call_create = netns.create().call_args_list
+        call_remove = netns.remove().call_args_list
+        nslist_create = set([x[0][0] for x in call_create if x[0]])
+        nslist_remove = set([x[0][0] for x in call_remove if x[0]])
+
+        assert calls == 1
+        assert nsptr is not None
+        assert nsptr in nslist_create
+        assert nsptr in nslist_remove
+    '''
+    nsname = uifname()
+    netns.create(nsname)
+    with IPRoute(netns=nsname) as ipr:
+        ipr.link('set', index=1, state='up')
+        ipr.poll(ipr.addr, 'dump', address='127.0.0.1', timeout=5)
```

Comment:
Oh, if this does what i think it does, it's really interesting & i did not know about it ! (maybe add a comment ?)

Reply:
It waits for a match in the dump output.

https://docs.pyroute2.org/iproute_linux.html#pyroute2.iproute.linux.RTNL_API.poll